node/engine/chainservice/simulated_backend_chainservice_test.go

```diff
@@ -147,6 +150,34 @@ func TestSimulatedBackendChainService(t *testing.T) {
 	if !bytes.Equal(statusOnChain[:], emptyBytes[:]) {
 		t.Fatalf("Adjudicator not updated as expected, got %v wanted %v", common.Bytes2Hex(statusOnChain[:]), common.Bytes2Hex(emptyBytes[:]))
 	}
+
+	// Start new chain service. It should detect old chain events that were emitted while it was offline
+	cs2, err := NewSimulatedBackendChainService(sim, bindings, ethAccounts[1])
+	defer closeChainService(t, cs2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	sim.Commit()
+
+	var receivedEvent Event
+	for i := 0; i < 2; i++ {
+		receivedEvent = <-cs2.EventFeed()
+		_, ok := receivedEvent.(DepositedEvent)
+		if !ok {
+			t.Fatalf("Expected chain event to be DepositedEvent")
+		}
+	}
+
+	receivedEvent = <-cs2.EventFeed()
+	if diff := cmp.Diff(expectedConcludedEvent, receivedEvent, cmp.AllowUnexported(ConcludedEvent{}, commonEvent{})); diff != "" {
+		t.Fatalf("Received event did not match expectation; (-want +got):\n%s", diff)
+	}
+
+	receivedEvent = <-cs2.EventFeed()
```

Comment:
Would it be worth dispatching a new event and making sure that our `cs2` also successfully picks up new blocks after starting late?

Reply:
Yes that would be a more fully-featured test. Will add that

Reply:
Added in https://github.com/statechannels/go-nitro/pull/1609/commits/c867cef4d4b2e791b9430b93e4aed52f84406252